### PR TITLE
Malaysia Vaccination Metrics App  for react redux capstone  project

### PR DESCRIPTION
**Pull request by Armynerh**
Selected the API  to use .
Built file structure for React app.
Prepared routes and navigation in the app.
Made sure that a user can display a list of items and filter them. Data comes from the API.
Created the tests for the application.
Styled components to match the design provided.
Deployed the project and tested for final details.

Recorded the 4th video for the project. https://www.loom.com/share/47fd0194e11e4b54b03e138909d28876?sid=258cfaef-5f62-4610-9225-b88bda424da8

Created a good README and PR description.

### CODE REVIEW

File: src/component/Metrics.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+              className="sr-only"
+              aria-live="polite"
+            />
+          </form>
+
+        </div>
+      </div>
+      <div className="d-flex flex-wrap m-auto justify-content-around">
+        {met.map((metric, i) => (
+          <div key={metric.total} className="col-md-4 col-lg-3 p-5 m-3 shadow" style={i % 2 === 0 ? { backgroundColor: '#4268B0' } : { backgroundColor: '#4063A7;' }}>
```
Comment (descholar-ceo) [inline src/component/Metrics.js]: - [ ] Would you remove a semicolon from this style? It's currently throwing an error on the console

File: package.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
+{
+  "name": "malaysia_vacination_metrics",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@reduxjs/toolkit": "^1.9.5",
+    "@testing-library/react": "^13.4.0",
+    "@testing-library/user-event": "^13.5.0",
```
Comment (descholar-ceo) [inline package.json:8]: - [ ] [Optional] You don't need to put these packages in project's dependencies, they for development environment, just put them in devDependencies

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @Armynerh 🤝 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- The project is deployed using Render properly, the home and detail page works as expected.
- The project styling is really commendable with great responsiveness
- The main purpose of the project has been achieved by having an index page, details page, and a filtering functionality 
- I like the most how you added the loading functionality, this improves the user experience

## Required Changes ♻️
_Check the comments under the review._

- [x] Even though you highlight that the most interesting part of this project was choosing the API to use, the project’s specs require that you highlight a piece of code that interested you and you feel proud of the most, would you please re-record the video and showcase that part of the code you feel proud of?
- [x] I would advise you to make a descriptive PR title, but shorter than what you have currently, you are showing the URL in the PR title but that makes it unreadable and hard to know what is the PR about
- [x] The Acknowledgement in readme file should give appropriate credit to [Nelson Sakwa](https://www.behance.net/sakwadesignstudio) the [author of the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/).
- [x] Components: ErrorPage and Metrics should have test cases at least
- [x] There is also a small issue with responsiveness in small screens (320px of width) on the details page, the cards are very small and letters are going off-card, you need to have a look into that as well
Screenshot 2023-09-14 at 19 51 51

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/component/Metrics.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+              className="sr-only"
+              aria-live="polite"
+            />
+          </form>
+
+        </div>
+      </div>
+      <div className="d-flex flex-wrap m-auto justify-content-around">
+        {met.map((metric, i) => (
+          <div key={metric.total} className="col-md-4 col-lg-2  p-3 m-3 shadow" style={i % 2 === 0 ? { backgroundColor: '#4268B0' } : { backgroundColor: '#4063A7;' }}>
```
Comment (binhussen) [inline src/component/Metrics.js]: - Kindly, remove the semicolon from `backgroundColor: '#4063A7;'` to fix the console error as suggested by the previous reviewer

File: README.md [hunk 4]
```diff
@@ -147,7 +147,9 @@ If you like this project Give Me a Star
 ## 🙏 Acknowledgments <a name="acknowledgements"></a>
 
 
-I would like to thank Microverse for the learning Experience
+I will like to thank Microverse for the learning Experience.
```
Comment (binhussen) [inline README.md:150]: - Great Job for updating the video record, Even though you forgot to highlight a piece of code that interested you and you feel proud of the most, would you please re-record the video and showcase that part of the code you feel proud of as suggested by previous reviewer 👍

**Review by binhussen — CHANGES_REQUESTED** [review]
Hi @Armynerh , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


## Highlights

- Sticked Given Design✔️
- Well Structured Code ✔️
- Used Git flow ✔️


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 4 above)
Comment (Armynerh) [reply README.md:150]: @binhussen  Thank you for your respponse, However, as the last reviewer suggested I remade the video and  clearly highlighte not 1 but multiple things I found interesting in the project.

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 🚧 

Hi @Armynerh 👋🏼 
I'm Andres.

You've done a great job implementing some of the requested changes for this milestone 🚀, but there are still some which haven't been addressed yet. You're getting closer, keep up the good work! 🤖 🎆 



 ## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 


 ## Required Changes ♻️
_Check the comments under the review._

- [ ] Although you did a great presentation, what the previous reviews are asking about the video, and according to the requirements, you should highlight a piece of code that you're proud of. In the video, you spoke about features that made you proud, but you didn't show us 
the code behind the feature. Please consider adding this to your video! 🤓 🔧 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Kenny,  

**React Capstone project** is complete! 🏁 Great job, congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job on introducing improvement to the presentation of the project!

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ahmetbozaci** [comment]
### Hello @Armynerh 
### _This is an additional review as a result of the second opinion request._ 

- Well done to implement the requirement.
_Even if some students were approved without following it, it's likely because the code reviewer missed it. The [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md#:~:text=you%20should%20also%20highlight%20some%20interesting%20piece%20of%20code%20or%20something%20you%20built%20that%20you%20are%20very%20proud%20of.) is straightforward: briefly showcase and explain a specific part of your project's code that you're proud of or that demonstrates your coding skills effectively._
___



### Cheers and Happy coding! 💻 👏
____
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Pull request merged**